File: src/parse.js
/**
 * Parse participants from slash command text.
 * Supports @mentions (extracted as Slack user IDs) and plain text names.
 * e.g. "/standup @alice j r" → ['U123ABC', 'j', 'r']
 */
function parseParticipants(text) {
  const participants = [];
  // Replace mentions with their UID and collect them
  const withoutMentions = text.replace(/<@([A-Z0-9]+)(?:\|[^>]+)?>/g, (_, uid) => {
    participants.push(uid);
    return '';
  });
  // Remaining whitespace-separated tokens are plain names
  for (const token of withoutMentions.trim().split(/\s+/)) {
    if (token) participants.push(token);
  }
  return participants;
}

/** Extract only Slack user IDs from @mentions (for add/remove subcommands). */
function parseMentionsOnly(text) {
  return [...text.matchAll(/<@([A-Z0-9]+)(?:\|[^>]+)?>/g)].map((m) => m[1]);
}

module.exports = { parseParticipants, parseMentionsOnly };

File: internal/parse/parse.go
package parse

import (
	"regexp"
	"strings"
)

var mentionRe = regexp.MustCompile(`<@([A-Z0-9]+)(?:\|[^>]+)?>`)

// ParseParticipants extracts Slack UIDs from <@UID> / <@UID|name> mentions,
// then collects remaining whitespace-separated tokens as plain names.
func ParseParticipants(text string) []string {
	var participants []string

	// Extract mentions and remove them from text
	withoutMentions := mentionRe.ReplaceAllStringFunc(text, func(match string) string {
		m := mentionRe.FindStringSubmatch(match)
		if m != nil {
			participants = append(participants, m[1])
		}
		return ""
	})

	// Remaining whitespace-separated tokens are plain names
	for _, token := range strings.Fields(withoutMentions) {
		if token != "" {
			participants = append(participants, token)
		}
	}

	if participants == nil {
		return []string{}
	}
	return participants
}

// ParseMentionsOnly extracts only Slack user IDs from <@...> mentions.
func ParseMentionsOnly(text string) []string {
	matches := mentionRe.FindAllStringSubmatch(text, -1)
	result := make([]string, 0, len(matches))
	for _, m := range matches {
		result = append(result, m[1])
	}
	return result
}
